### PR TITLE
Binfmt fix

### PR DESCRIPTION
**Pull request by jeongchanKim**

### CODE REVIEW

File: os/binfmt/Kconfig [hunk 1]
```diff
@@ -90,6 +90,16 @@ config OPTIMIZE_APP_RELOAD_TIME
 		the MPU requirements. Hence, using this option can result in a steep increase in the memory
 		requirement for this application.
 
+config BINFMT_SECTION_UNIFIED_MEMORY
+	bool "Allocate section memory as one chunk"
+	default n
```
Comment (sunghan-chang) [inline os/binfmt/Kconfig:95]: This is valid with reload optimization config enabled. Could you make a dependency with that?

File: os/binfmt/Kconfig [hunk 2]
```diff
@@ -90,6 +90,16 @@ config OPTIMIZE_APP_RELOAD_TIME
...
+	---help---
+		When this option is enabled, binary loader allocates one big memory chunk enough to contains each loading sections like text, ro, data
+		and uses each loading sections address by pointer.
```
Comment (sunghan-chang) [inline os/binfmt/Kconfig:98]: Could you add one more line?
```
When this option is disabled, by default, loader allocates separated memory for each section.

File: os/binfmt/binfmt_dumpmodule.c [hunk 3]
```diff
@@ -91,8 +91,9 @@ int dump_module(FAR const struct binary_s *bin)
 		berr("  argv:      %p\n", bin->argv);
 		berr("  entrypt:   %p\n", bin->entrypt);
 		berr("  mapped:    %p size=%d\n", bin->mapped, bin->mapsize);
-		berr("  alloc:     %p %p %p\n", bin->alloc[ALLOC_TEXT], bin->alloc[ALLOC_CTOR], bin->alloc[ALLOC_DTOR]);
+		berr("  text alloc:     %p\n", bin->alloc[ALLOC_TEXT]);
```
Comment (sunghan-chang) [inline os/binfmt/binfmt_dumpmodule.c:94]: 1. Should this be text? Not to be start of alloc?
2. don't you need to add ro and data?
These queries could be treated in another pr.

File: os/binfmt/binfmt_unloadmodule.c [hunk 4]
```diff
@@ -188,13 +189,32 @@ int unload_module(FAR struct binary_s *binp)
 
 		/* Free allocated address spaces */
 
-#ifndef CONFIG_APP_BINARY_SEPARATION
-		if (binp->alloc[i]) {
-			binfo("Freeing alloc[%d]: %p\n", i, binp->alloc[i]);
-			kmm_free((FAR void *)binp->alloc[i]);
-		}
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+		if (binp->reload != true) {
```
Comment (sunghan-chang) [inline os/binfmt/binfmt_unloadmodule.c:193]: Let's do like below
```
#ifdef config_opti...
    if (....)
#endif 
    {
           .......
     }
```

File: os/binfmt/Kconfig (hunk 1 above)
Comment (jeongchanKim) [reply os/binfmt/Kconfig:96]: Done.

File: os/binfmt/binfmt_dumpmodule.c (hunk 3 above)
Comment (jeongchanKim) [reply os/binfmt/binfmt_dumpmodule.c:97]: Modified. Thanks.

File: os/binfmt/binfmt_unloadmodule.c (hunk 4 above)
Comment (jeongchanKim) [reply os/binfmt/binfmt_unloadmodule.c:214]: Done. Thanks.

File: os/binfmt/Kconfig [hunk 5]
```diff
@@ -90,6 +90,18 @@ config OPTIMIZE_APP_RELOAD_TIME
 		the MPU requirements. Hence, using this option can result in a steep increase in the memory
 		requirement for this application.
 
+config BINFMT_SECTION_UNIFIED_MEMORY
+	bool "Allocate section memory as one chunk"
+	depends on OPTIMIZE_APP_RELOAD_TIME
+	default n
+	---help---
+		When this option is enabled, binary loader allocates one big memory chunk enough to contains each loading sections
```
Comment (jeongarmy) [inline os/binfmt/Kconfig:98]: Need to change `contains` to `contain`

Comment (jeongchanKim) [reply os/binfmt/Kconfig:98]: Why? The subject of contains is `one big memory chunk`, and it is the singular.

**Review by sunghan-chang — APPROVED** [review]
